src/Playwright.Tests/BrowserTypeConnectTests.cs

```diff
@@ -80,71 +82,396 @@ public void TearDown()
         {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            IBrowser disconneced = null;
+            browser.Disconnected += (_, e) => disconneced = e;
+            await browser.CloseAsync();
+            Assert.AreEqual(browser, disconneced);
         }
 
         [PlaywrightTest("browsertype-connect.spec.ts", "should set the browser connected state")]
-        [Ignore("SKIP WIRE")]
-        public void ShouldSetTheBrowserConnectedState()
+        public async Task ShouldSetTheBrowserConnectedState()
         {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            Assert.AreEqual(browser.IsConnected, true);
+            await browser.CloseAsync();
+            Assert.AreEqual(browser.IsConnected, false);
         }
 
         [PlaywrightTest("browsertype-connect.spec.ts", "should throw when used after isConnected returns false")]
-        [Ignore("SKIP WIRE")]
-        public void ShouldThrowWhenUsedAfterIsConnectedReturnsFalse()
+        public async Task ShouldThrowWhenUsedAfterIsConnectedReturnsFalse()
+        {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var page = await browser.NewPageAsync();
+            await browser.CloseAsync();
+            Assert.AreEqual(browser.IsConnected, false);
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(async () => await page.EvaluateAsync("1 + 1"));
+            StringAssert.Contains("Connection closed (Browser closed)", exception.Message);
+        }
+
+        [PlaywrightTest("browsertype-connect.spec.ts", "should throw when calling waitForNavigation after disconnect")]
+        public async Task ShouldThrowWhenWhenCallingWaitForNavigationAfterDisconnect()
         {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var page = await browser.NewPageAsync();
+            await browser.CloseAsync();
+
+            Assert.AreEqual(browser.IsConnected, false);
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(async () => await page.WaitForNavigationAsync());
+            StringAssert.Contains("Page closed", exception.Message);
         }
 
         [PlaywrightTest("browsertype-connect.spec.ts", "should reject navigation when browser closes")]
-        [Ignore("SKIP WIRE")]
-        public void ShouldRejectNavigationWhenBrowserCloses()
+        public async Task ShouldRejectNavigationWhenBrowserCloses()
         {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var page = await browser.NewPageAsync();
+            await browser.CloseAsync();
+
+            Assert.AreEqual(browser.IsConnected, false);
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(async () => await page.GotoAsync(Server.EmptyPage));
```

Comment:
This is not what the original test does.

Reply:
Have changed it accordingly